### DISCUSSION TITLE
What should a forked Spectre project look like?

### DISCUSSION THREAD

**Discussion by davestewart** [spectre-org/spectre-css#2] [Ideas]
I thought this would be a good place to continue the discussion from [the original PR](https://github.com/picturepan2/spectre/pull/685#issuecomment-1726646030).

In that thread  I suggested creating an org and fork to ensure Spectre CSS is available going forwards.

Questions around the feasibility of that include:

- what would a forked repo / org look like?
- what needs to be done now?
- what would be good to do soon?
- what would be nice to do in future? 

I'm looping in recent PR authors and contributors:

> @ma4nn @Valexr @codebling @peterhil @blokhin @jviotti @salamwaddah @janbakker2525 @NazimHAli @janbakker2525 @akirk @McLaynV @mlbiche @matronator @NicoHood @wyndon @st-wong @endorama @marcosstefani @Vectrex @nisarhassan12

Here are my thoughts...

Need to do now:

- [x] rename the org (see #3)
- [x] fork the original code
- [x] publish [docs](https://spectre-org.github.io/spectre-css/) 
- [ ] review outstanding PRS (see #5)
- [x] publish a new package (in the meantime, can `npm i spectre-org/spectre-css`)

Good to do soon:

- [x] remove support for LESS
- [x] proper markdown docs
- [x] a `@spectre-org` NPM namespace

Nice to do in future:

- [ ] introduce theming
- [ ] wrappers for Vue, React, etc

Who knows if this will go somewhere, but drop your thoughts and suggestions below.

**Comment by ma4nn** [comment]
@davestewart thanks for your effort, good idea with the new organization 👍 

A couple of thoughts here:

- I don't think the org name "picturepan-too" is very appropriate (because "picturepan2" is the original author), imho it should better contain the framework name (e.g. spectre-org) or kind of a successor framework name (e.g. spectre2, spectre-fork, skyfall-css, ..)
- We should define what is the purpose of this fork, i.e. which PRs to accept, what is the direction for new developments (e.g. I personally would keep it simple and extremely lightweight as it currently is, i.e. no theming, wrappers, etc. within the original repo), etc.
- Also we should discuss how/who will maintain it (admins, owners, etc.) so there won't be a situation like in the original repo/org

**Reply by davestewart** [reply]
Hey @ma4nn,

Thanks for the vote of confidence!

**Org name**

I see your point regarding the org name; I thought it was a fun nod to the creator, but can see it might be seen as controversial.

I've checked the settings, and we can rename it, so have started a new discussion to vote on a name:

- #3 

**Purpose**

Agreed on PRs.

As for wrappers, I'd be more than happy to wrap Spectre into a set of Vue wrappers and create a new repo `vue-spectre` or such like. Perhaps others are already part-way down the React equivilents?

Regards other packages, note there is also a [Spectre Icons repo](https://picturepan2.github.io/spectre/elements/icons.html).

Theming I only suggested as I've made some simple updates in some of my project which make it look a little more modern. It already supports variables, so perhaps it's just a tweak on those.

Or, perhaps a more Tailwind-like config file would be a good move.

**Maintainers**

Totally agreed. Happy to take suggestions.

**Reply by NazimHAli** [reply]
+1 about library name! Before making any changes, it would be good to set up basic things like automated linting, testing, metrics, publishing, deploying on every commit for PRs and releases.

Keeping it lightweight is ideal and we can keep those separate from the core library.

**Reply by davestewart** [reply]
OK, renames done; we now have:

```
spectre-org/spectre-css
```

**Reply by ma4nn** [reply]
> Maintainers
> Totally agreed. Happy to take suggestions.

Hi @davestewart back from vacation now🍹I saw you committed and merged already several changes, nice 👍 

Can you please elaborate a bit more on what is your plan regarding maintainers, review processes, priorization, releases, etc. I'm more than ready to invest time in this project and make spectre-css alive again, but the common goal should be clear. Also regarding Admin rights on the repositories/org, etc.

In addition I very much support the suggestion from @NazimHAli that we should add linting, testing, etc. in the first place before applying any changes.

Happy to discuss this also directly, e.g. via Skype, Teams, Discord, whatever 😉 Just let me know.

**Reply by davestewart** [reply]
Hey! Been a busy few weeks for myself, how are you fixed this week?

DM me on Twitter and we can coordinate @dave_stewart

**Reply by ma4nn** [reply]
@davestewart seems like I cannot send you DMs without being a Premium Member🙈 can you follow my account @vianetz then it should be possible to exchange DMs

**Comment by codebling** [comment]
Thanks for tagging me! Don't think I have an interest in contributing/maintaining a lib that I'm not actively using at the moment, but thanks for making me part of the conversation!

**Comment by st-wong** [comment]
Not sure if I have the time to contribute, but definitely looking forward to seeing how things go! All the best!

**Comment by davestewart** [comment]
If anyone is curious, I've migrated the existing html docs to Markdown, and have set them up in VitePress:

![image](https://github.com/spectre-org/spectre-css/assets/132681/39e34441-86b8-4827-b895-907243f4dae7)

It's live here:

- https://spectre-org.github.io/spectre-docs/

There are a few bits to finish off:

- https://github.com/spectre-org/spectre-docs/issues/4
- https://github.com/spectre-org/spectre-docs/issues/7

Create issues if you need to here:

- https://github.com/spectre-org/spectre-docs/issues

**Comment by endorama** [comment]
@davestewart thanks for the ping! I've not been working with front-end things for a while now, so I'm a bit out of touch.

I like the direction, I cannot commit to a maintenance schedule but I'll look around for opportunity to help (i.e. reviewing outstanding PRs). I agree with the current work in progress and suggestions in the thread created by @ma4nn.

**Reply by davestewart** [reply]
Cool, thanks for the reply!

**Comment by peterhil** [comment]
Great! I use Spectre.css actively at work and on hobby projects. I hope I can help somehow. A least I could review some pull requests.

**Comment by davestewart** [comment]
Just to let everyone know, Spectre CSS is now on NPM!

- https://www.npmjs.com/package/@spectre-org/spectre-css

We have also updated the docs with simpler install instructions, search, and better information around customisation.

- https://spectre-org.github.io/spectre-docs/

In the coming weeks and months we have modest plans to make Spectre CSS even better.

Issues:

- https://github.com/spectre-org/spectre-css/issues
- https://github.com/spectre-org/spectre-docs/issues

Projects:

- https://github.com/orgs/spectre-org/projects

**Reply by akirk** [reply]
Looks fantastic!

**Reply by davestewart** [reply]
Thanks! I think so too. Really excited to breathe new life back into such a great project
